src/dmd/dmodule.d

```diff
@@ -1356,3 +1383,263 @@ struct ModuleDeclaration
+{
+    static size_t findSortedIndexToAddForDepth(size_t depth)
+    {
+        size_t index = 0;
+        while (index < matchNodes.dim)
+        {
+            auto info = matchNodes[index];
+            if (depth > info.depth)
+                break;
+            index += 1 + info.depth;
+        }
+        return index;
+    }
+
+    if (matchNodes.dim == 0)
+    {
+        for (size_t packageIndex = 0; packageIndex < global.params.includeMatchStrings.dim; packageIndex++)
+        {
+            auto matchString = global.params.includeMatchStrings[packageIndex];
+            auto depth = parseMatchStringDepth(matchString);
+            auto entryIndex = findSortedIndexToAddForDepth(depth);
+            matchNodes.split(entryIndex, depth + 1);
+            parseMatchString(matchString, &matchNodes[entryIndex], depth);
+            // if at least 1 "include pattern" is given, then it is assumed the
+            // user only wants to include modules that were explicitly given, which
+            // changes the default behavior from inclusion to exclusion.
+            if (includeByDefault && !matchNodes[entryIndex].isExclude)
+            {
+                //fprintf(global.stdmsg, "Matcher: found 'include pattern', switching default behavior to exclusion\n");
+                includeByDefault = false;
+            }
+        }
+
+        // Add the default 1 depth matchers
+        MatcherNode[8] defaultDepth1MatchNodes = [
+            MatcherNode(true, 1), MatcherNode(Id.std),
+            MatcherNode(true, 1), MatcherNode(Id.core),
+            MatcherNode(true, 1), MatcherNode(Id.etc),
+            MatcherNode(true, 1), MatcherNode(Id.object),
+        ];
+        {
+            auto index = findSortedIndexToAddForDepth(1);
+            matchNodes.split(index, defaultDepth1MatchNodes.length);
+            matchNodes.data[index .. index + defaultDepth1MatchNodes.length] = defaultDepth1MatchNodes[];
+        }
+    }
+}
+
+/**
+ * Pattern strings are taken directly from the -i command line option,
+ * so they can either end with the traditional terminating null, or they
+ * can end with a comma.  This function returns true if the given character
+ * matches either of these.
+ * Returns:
+ *  True if the given character is '\0' or ','.
+ */
+private bool endOfPattern(char c) { return c == ',' || c == '\0'; }
+
+/**
+ * Determines the depth of the given match string.
+ * Params:
+ *  matchString = The match string to determine the depth of.
+ * Returns:
+ *  The component depth of the given match string.
+ */
+private ushort parseMatchStringDepth(const(char)* matchString)
+{
+    if (matchString[0] == '-')
+        matchString++;
+
+    // handle special case
+    if (matchString[0] == '.' && endOfPattern(matchString[1]))
+        return 0;
+
+    ushort depth = 1;
+    for (;; matchString++)
+    {
+        auto c = *matchString;
+        if (c == '.')
+            depth++;
+        if (endOfPattern(c))
+            return depth;
+    }
+}
```

Comment:
We have a `unittest` target in the Makefile since two weeks and test it on auto-tester.
Also inline unittests greatly help the reader:

```d
unittest
{
   assert("-".parseMatchStringDepth == 0);
   assert("-.".parseMatchStringDepth == 0);
   assert("-abc".parseMatchStringDepth == 1);
   assert("-abc.foo".parseMatchStringDepth == 2);
   assert("-abc -abc".parseMatchStringDepth == 1);
   assert("-abc-abc".parseMatchStringDepth == 1);
}
```
  
The other new functions in this module would greatly profit from this too!